### ISSUE TITLE
provisioner/chef: find -maxdepth not supported on AIX

### ISSUE THREAD

**Issue by cattywampus** [hashicorp/terraform#15707]
When I'm using terraform to build virtual machines running AIX, I get an error during the chef provisioning step. Details of the error are in the report below, but it looks like the linux chef provisioner has a step where it uses `find -maxdepth 1` to find all of the config files in `/etc/chef` and change the permission bits.

The problem is the `-maxdepth` option is not supported in the version of the `find` command on AIX. I ran into this on AIX 7.1, and from my basic research this would apply to AIX 6.1 too. 

I think the fix for this would be to alter the find command to return the same list of files from that directory (and no deeper) without using `-maxdepth`. I found a Stackoverflow answer that provides an [alternative technique](https://unix.stackexchange.com/questions/372994/equivalent-maxdepth-for-find-in-aix) to using `find` without the `-maxdepth` option. I modified this solution to match what the linux chef provisioner does and tested this command in AIX 7.1 and RedHat 7.1 with success. 

If this is a bug you would be willing to accept, I'd be happy to submit a pull request for it. 

### Terraform Version
Terraform v0.9.2

I understand this isn't the latest version but `-maxdepth` flag is still being used in [master](https://github.com/hashicorp/terraform/blob/master/builtin/provisioners/chef/linux_provisioner.go#L13) and [v0.10.0](https://github.com/hashicorp/terraform/blob/v0.10.0/builtin/provisioners/chef/linux_provisioner.go#L13).

### Terraform Configuration Files
I'm currently waiting on approval to post a sanitized version of the terraform plan, but until then I can say that I have a `openstack_compute_instance_v2` resource with two provisioners:
  * `remote-exec` - downloads the chef `install.sh` script from an internal server and runs it to install the chef-client. This definitely finishes successfully. The chef client is installed. 
  * `chef` - performs the steps to actually register the node with the chef server, set the run list and environment. `skip_install` is set to `true` because chef-client is installed by the previous `remote-exec` provisioner.

### Debug Output
I can't provide a complete debug output yet due to pending approval through our security process. Once I get the 👍 then I'll add the link to a Gist for the relevant section of the debug log output that fails during the chef provisioner process. 

In the interim, I'm hoping the error below in the [Actual Behavior](#actual-behavior) section is enough to kick start this discussion since the error is pretty cut and dry. 

### Panic Output
Not applicable

### Expected Behavior
When I create a virtual machine (currently using openstack with an AIX template) the VM should be provisioned with Chef so that it's managed by my Chef Server, assigned the proper run list and environment, and successfully checks in with the Chef Server. 

### Actual Behavior
When the chef provisioner runs I can see from the DEBUG output that it is able to run most of the commands but fails when trying to modify the permissions of the `/etc/chef` files. An error occurs that says:

```
* Command "sudo find /etc/chef -maxdepth 1 -type f -exec /bin/chmod 666 {} +" exited with non-zero exit status: 1
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a plan that includes a VM built with AIX as the operating system
2. Add a chef provisioner to the plan
3. `terraform apply`

### Important Factoids
The find command on AIX 6.1 and 7.1 does not support the `-maxdepth` option. 

### References
No related issues or pull requests in this project that I could find.

**Comment by surajsub** [comment]
Has this been fixed in the 0.11.3 version of TF ? I'm running into the same issue with chef on RedHat 7.

**Comment by danieldreier** [comment]
I'm closing this issue because we announced tool-specific (vendor or 3rd-party) provisioner deprecation in [mid-September 2020](https://discuss.hashicorp.com/t/notice-terraform-to-begin-deprecation-of-vendor-tool-specific-provisioners-starting-in-terraform-0-13-4/13997). Additionally, we added a deprecation notice for tool-specific provisioners in 0.13.4. On a practical level this means we will no longer be reviewing or merging PRs for built-in plugins like the chef provisioner.

The discuss post linked above explains this in more depth, but the basic reason we're making this change is that these vendor provisioners have been extremely challenging for us to maintain, and are a weak spot in the terraform user experience. People reach for them not realizing the bugs and UX limitations, and they're areas that are difficult for us to maintain because of the huge surface area of integrating with a bunch of different tools (Puppet, Chef, Salt, etc) that each require deep domain knowledge to do right. For example, testing each of these against all the versions of those tools, on multiple platforms, is prohibitive, and so we don't - but users have a reasonable expectation that everything in the Terraform Core codebase is well tested. Similarly, it's tough to accept PRs, even for useful improvements, because we don't have anyone on the core team with deep Chef knowledge, and we have not been able to get community volunteers to own PR review for this codebase, so it's a shot in the dark whether a given PR makes things better or worse from the perspective of an experienced Chef + Terraform user.

For the time being, the best option if you want to fix this bug, is to work with the community and build a standalone chef provisioner, fix this in it, and distribute it as a plugin binary, similar to how the [ansible provisioner](https://github.com/radekg/terraform-provisioner-ansible) is distributed.

I'm aware of the limitations of this approach, but it's the best option compared to coupling provisioner development to the Terraform Core release lifecycle. We believe the benefit to users of having provisioner development decoupled from core, exceeds the convenience of having these provisioners built in to core. We want to provide a better user experience in the future, and our hope here is that the ability to improve, fix and repair provisioners without us blocking their development, much like providers, will help make a strong case for what's next.

I think it’s also important to highlight that we have no plans to remove the generic provisioners or the pluggable functionality during Terraform's 1.0 lifecycle. 

I appreciate your input here to improve Terraform, and am always happy to talk. Please feel free to reach out to me or Petros Kolyvas if you would like to talk more about this change.
